src/Compilers/Test/Utilities/CSharp/NullableAttributesVisitor.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp.Test.Utilities
+{
+    /// <summary>
+    /// Returns a string with all symbols containing nullable attributes.
+    /// </summary>
+    internal sealed class NullableAttributesVisitor : CSharpSymbolVisitor
+    {
+        internal static string GetString(Symbol symbol)
+        {
+            var builder = new StringBuilder();
+            var visitor = new NullableAttributesVisitor(builder);
+            visitor.Visit(symbol);
+            return builder.ToString();
+        }
+
+        private readonly StringBuilder _builder;
+        private readonly HashSet<Symbol> _reported;
+
+        private NullableAttributesVisitor(StringBuilder builder)
+        {
+            _builder = builder;
+            _reported = new HashSet<Symbol>();
+        }
+
+        public override void DefaultVisit(Symbol symbol)
+        {
+            ReportSymbol(symbol);
+        }
+
+        public override void VisitModule(ModuleSymbol module)
+        {
+            Visit(module.GlobalNamespace);
+        }
+
+        public override void VisitNamespace(NamespaceSymbol @namespace)
+        {
+            VisitList(@namespace.GetMembers());
+        }
+
+        public override void VisitNamedType(NamedTypeSymbol type)
```

Comment:
>VisitNamedType [](start = 29, length = 14)

Should we visit constraints?

Reply:
Visiting constraints doesn't fit easily into this model since we're assuming all symbols are definitions.

---
In reply to: [293585452](https://github.com/dotnet/roslyn/pull/36398#discussion_r293585452) [](ancestors = 293585452)